codalab/rest/legacy.py

```diff
@@ -447,6 +469,23 @@ def get_worksheet_content(uuid):
     service = BundleService()
     return service.full_worksheet(uuid)
 
+# Needs to store the last unfinished bundle uuids because if a bundle just finished running,
+# unfinished_bundle_uuids won't include that bundle (from the backend points of view, it's a finished bundle)
+# whereas the frontend is still waiting to update its state from running to ready/failed.
+last_unfinished_bundle_uuids = set()
+# Get items that include unfinished run bundles in a worksheet, uuid is the uuid for that worksheet
+@get('/api/worksheets/unfinished_bundles/<uuid:re:%s>/' % spec_util.UUID_STR)
+def get_worksheet_content_with_unfinished_bundles(uuid):
+    global last_unfinished_bundle_uuids
+    service = BundleService()
+    unfinished_bundle_uuids = [bundle.uuid for bundle in local.model.batch_get_bundles(state=State.OPTIONS - State.FINAL_STATES)]
```

Comment:
I think the frontend should just pass in the unfinished bundles and then this method just pass back the info for them.  In fact, this method should have nothing to do with finished/unfinished - all it does is just pass back information for an arbitrary subset of bundles.  Can we just include this list as query params for the existing get_worksheet_content endpoint?